### PR TITLE
(dsl): Support prefix query

### PR DESCRIPTION
**Pull request by petarcurcin**

### CODE REVIEW

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 1]
```diff
@@ -544,6 +544,38 @@ object ElasticQuery {
...
 
+  /**
+   * Constructs a type-safe instance of [[zio.elasticsearch.query.PrefixQuery]] using the specified parameters.
+   * [[zio.elasticsearch.query.PrefixQuery]] is used for matching documents that contain a specific prefix in a provided
+   * field.
+   *
+   * @param field
+   *   the type-safe field for which query is specified for
+   * @param value
+   *   the value that will be used for the query, represented by an instance of type `A`
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:555]:
```suggestion
   *   the text value that will be used for the query
```

Comment (dbulaja98) [reply modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:555]: Can you replace this for term method also?

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 2]
```diff
@@ -544,6 +544,38 @@ object ElasticQuery {
...
+
+  /**
+   * Constructs an instance of [[zio.elasticsearch.query.PrefixQuery]] using the specified parameters.
+   * [[zio.elasticsearch.query.PrefixQuery]] is used for matching documents that contain a specific prefix in a provided
+   * field.
+   *
+   * @param field
+   *   the field for which query is specified for
+   * @param value
+   *   the value that will be used for the query, represented by an instance of type `A`
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:572]: Same here.

File: modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala [hunk 3]
```diff
@@ -665,6 +665,25 @@ private[elasticsearch] case object Unbounded extends LowerBound with UpperBound
...
 
+sealed trait PrefixQuery[S] extends ElasticQuery[S] with HasCaseInsensitive[PrefixQuery[S]]
+
+private[elasticsearch] final case class Prefix[S](
+  field: String,
+  value: String,
+  caseInsensitive: Option[Boolean]
+) extends PrefixQuery[S] { self =>
+
+  override def caseInsensitive(value: Boolean): PrefixQuery[S] =
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala:676]: Can you omit `override`?

File: modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala [hunk 4]
```diff
@@ -665,6 +665,25 @@ private[elasticsearch] case object Unbounded extends LowerBound with UpperBound
...
+    self.copy(caseInsensitive = Some(value))
+
+  override private[elasticsearch] def toJson(fieldPath: Option[String]): Json = {
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala:679]: Here also.

File: modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala [hunk 5]
```diff
@@ -2531,6 +2531,50 @@ object ElasticQuerySpec extends ZIOSpecDefault {
           assert(queryWithScoreMode.toJson(fieldPath = None))(equalTo(expectedWithScoreMode.toJson)) &&
           assert(queryWithAllParams.toJson(fieldPath = None))(equalTo(expectedWithAllParams.toJson))
         },
+        test("prefix") {
+          val query                    = prefix(TestDocument.stringField, "test")
+          val queryWithCaseInsensitive = prefix(TestDocument.stringField, "test").caseInsensitiveTrue
+          val queryWithAllParams       = prefix(TestDocument.stringField, "test").caseInsensitiveFalse
```
Comment (dbulaja98) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala:2537]: Remove this check.

**Review by dbulaja98 — CHANGES_REQUESTED** [review]

**Review by markaya — DISMISSED** [review]

File: docs/overview/queries/elastic_query_prefix.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+---
+id: elastic_query_prefix
+title: "Prefix Query"
+---
+
+The `Prefix` query returns documents that contain a specific prefix in a provided field.
```
Comment (dbulaja98) [inline docs/overview/queries/elastic_query_prefix.md:6]:
```suggestion
The `Prefix` query returns documents that contain a specific prefix in the provided field.
```

File: modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala (hunk 5 above)
Comment (dbulaja98) [reply modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala]: We have only one parameter, so `queryWithAllParams` is same as `queryWithCaseInsensitive`.

**Review by dbulaja98 — DISMISSED** [review]

File: modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala [hunk 7]
```diff
@@ -2531,6 +2531,36 @@ object ElasticQuerySpec extends ZIOSpecDefault {
           assert(queryWithScoreMode.toJson(fieldPath = None))(equalTo(expectedWithScoreMode.toJson)) &&
           assert(queryWithAllParams.toJson(fieldPath = None))(equalTo(expectedWithAllParams.toJson))
         },
+        test("prefix") {
```
Comment (drmarjanovic) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala:2534]: Please provide `"prefix"` tests in `"constructing"` part as well.

**Review by drmarjanovic — CHANGES_REQUESTED** [review]

**Review by drmarjanovic — APPROVED** [review]

**Review by dbulaja98 — APPROVED** [review]

**Pull request merged**